std/random.d

```diff
@@ -1529,6 +1523,29 @@ version (AnyARC4Random)
     extern(C) private @nogc nothrow
     {
         uint arc4random() @safe;
+        void arc4random_buf(scope void* buf, size_t nbytes) @system;
+    }
+}
+else
+{
+    private ulong bootstrapSeed() @nogc nothrow
+    {
+        ulong result = void;
+        enum ulong m = 0xc6a4_a793_5bd1_e995UL; // MurmurHash2_64A constant.
+        void updateResult(ulong x)
+        {
+            x *= m;
+            x = (x ^ (x >>> 47)) * m;
+            result = (result ^ x) * m;
+        }
+        import core.thread : getpid, Thread;
+        import core.time : MonoTime;
+
+        updateResult(cast(ulong) cast(void*) Thread.getThis());
+        updateResult(cast(ulong) getpid());
+        updateResult(cast(ulong) MonoTime.currTime.ticks);
+        result = (result ^ (result >>> 47)) * m;
+        return result ^ (result >>> 47);
```

Comment:
With the LLVM optimizer, this whole function is optimized to calling `Thread.getThis()`, `getpid()` and `MonoTime.currTime().ticks()` (for potential side effects) and then returning an undefined value - due to the final value of `result` depending on its initial value, which is explicitly undefined.

The intent probably was to make the boostrap seed somewhat more random, but this doesn't work here. Would it be okay to initialize `result` with 0?

Reply:
Quoting Johan:
> Using an uninitialized value is UB in D. https://dlang.org/spec/declaration.html#void_init

Reply:
>Would it be okay to initialize `result` with 0?

Yes.